### ISSUE TITLE
Swift rollout卡住

### ISSUE THREAD

**Issue by Jianshu-Hu** [modelscope/ms-swift#4703]
在4090上跑命令行 CUDA_VISIBLE_DEVICES=0 swift rollout --model Qwen/Qwen2.5-VL-3B-Instruct会卡住。
**Package version:**
`python==3.10
torch==2.6.0
ms-swift==3.5.2
transformers==4.51.3
modelscope==1.27.1
peft==0.15.2
trl==0.18
deepspeed==0.16.9
vllm==0.8.5.post1`
**Terminal output:**
[2025-06-25 10:56:41,459] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:49: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(ctx, input, weight, bias=None):
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:67: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
Warning: The default cache directory for DeepSpeed Triton autotune, /home/nfs/jhu/.triton/autotune, appears to be on an NFS system. While this is generally acceptable, if you experience slowdowns or hanging when DeepSpeed exits, it is recommended to set the TRITON_CACHE_DIR environment variable to a non-NFS path.
 [WARNING]  async_io requires the dev libaio .so object and headers but these were not found.
 [WARNING]  async_io: please install the libaio-dev package with apt
 [WARNING]  If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
run sh: `/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/bin/python /bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/swift/cli/rollout.py --model Qwen/Qwen2.5-VL-3B-Instruct`
[2025-06-25 10:56:47,535] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:49: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(ctx, input, weight, bias=None):
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:67: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
[INFO:swift] Successfully registered `/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/swift/llm/dataset/data/dataset_info.json`.
Warning: The default cache directory for DeepSpeed Triton autotune, /home/nfs/jhu/.triton/autotune, appears to be on an NFS system. While this is generally acceptable, if you experience slowdowns or hanging when DeepSpeed exits, it is recommended to set the TRITON_CACHE_DIR environment variable to a non-NFS path.
 [WARNING]  async_io requires the dev libaio .so object and headers but these were not found.
 [WARNING]  async_io: please install the libaio-dev package with apt
 [WARNING]  If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
INFO 06-25 10:56:49 [importing.py:53] Triton module has been replaced with a placeholder.
INFO 06-25 10:56:49 [__init__.py:239] Automatically detected platform cuda.
[INFO:swift] rank: -1, local_rank: -1, world_size: 1, local_world_size: 1
[INFO:swift] Downloading the model from ModelScope Hub, model_id: Qwen/Qwen2.5-VL-3B-Instruct
[WARNING:modelscope] Using branch: master as version is unstable, use with caution
[INFO:modelscope] Target directory already exists, skipping creation.
[INFO:swift] Loading the model using model_dir: /bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2___5-VL-3B-Instruct
[INFO:swift] Setting torch_dtype: torch.bfloat16
[INFO:swift] args.result_path: /bd_byt4090i1/users/jhu/VLM-finetune/result/Qwen2.5-VL-3B-Instruct/deploy_result/20250625-105703.jsonl
[INFO:swift] Global seed set to 42
[INFO:swift] args: DeployArguments(model='Qwen/Qwen2.5-VL-3B-Instruct', model_type='qwen2_5_vl', model_revision=None, task_type='causal_lm', torch_dtype=torch.bfloat16, attn_impl=None, num_labels=None, problem_type=None, rope_scaling=None, device_map=None, max_memory={}, local_repo_path=None, init_strategy=None, template='qwen2_5_vl', system=None, max_length=None, truncation_strategy='delete', max_pixels=None, agent_template=None, norm_bbox=None, use_chat_template=True, padding_free=False, padding_side='right', loss_scale='default', sequence_parallel_size=1, response_prefix=None, template_backend='swift', dataset=[], val_dataset=[], split_dataset_ratio=0.01, data_seed=42, dataset_num_proc=1, load_from_cache_file=True, dataset_shuffle=True, val_dataset_shuffle=False, streaming=False, interleave_prob=None, stopping_strategy='first_exhausted', shuffle_buffer_size=1000, download_mode='reuse_dataset_if_exists', columns={}, strict=False, remove_unused_columns=True, model_name=None, model_author=None, custom_dataset_info=[], quant_method=None, quant_bits=None, hqq_axis=None, bnb_4bit_compute_dtype=torch.bfloat16, bnb_4bit_quant_type='nf4', bnb_4bit_use_double_quant=True, bnb_4bit_quant_storage=None, max_new_tokens=None, temperature=None, top_k=None, top_p=None, repetition_penalty=None, num_beams=1, stream=False, stop_words=[], logprobs=False, top_logprobs=None, ckpt_dir=None, lora_modules=[], tuner_backend='peft', train_type='lora', adapters=[], external_plugins=[], seed=42, model_kwargs={}, load_args=True, load_data_args=False, packing=False, packing_cache=None, custom_register_path=[], use_hf=False, hub_token=None, ddp_timeout=18000000, ddp_backend=None, ignore_args_error=False, use_swift_lora=False, tp=1, session_len=None, cache_max_entry_count=0.8, quant_policy=0, vision_batch_size=1, gpu_memory_utilization=0.9, tensor_parallel_size=1, pipeline_parallel_size=1, max_num_seqs=256, max_model_len=None, disable_custom_all_reduce=True, enforce_eager=False, limit_mm_per_prompt={}, vllm_max_lora_rank=16, enable_prefix_caching=False, use_async_engine=True, vllm_quantization=None, data_parallel_size=1, log_level='info', merge_lora=False, safe_serialization=True, max_shard_size='5GB', infer_backend='pt', result_path='/bd_byt4090i1/users/jhu/VLM-finetune/result/Qwen2.5-VL-3B-Instruct/deploy_result/20250625-105703.jsonl', write_batch_size=1000, metric=None, max_batch_size=1, val_dataset_sample=None, host='0.0.0.0', port=8000, api_key=None, ssl_keyfile=None, ssl_certfile=None, owned_by='swift', served_model_name=None, verbose=True, log_interval=20, max_logprobs=20)
Downloading Model from https://www.modelscope.cn to directory: /bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2.5-VL-3B-Instruct
[INFO:swift] Start time of running main: 2025-06-25 10:57:03.863206
[INFO:swift] swift.__version__: 3.5.2
INFO:     Started server process [1666264]
INFO:     Waiting for application startup.
[2025-06-25 10:57:08,183] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:49: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(ctx, input, weight, bias=None):
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:67: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
[INFO:swift] Successfully registered `/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/swift/llm/dataset/data/dataset_info.json`.
Warning: The default cache directory for DeepSpeed Triton autotune, /home/nfs/jhu/.triton/autotune, appears to be on an NFS system. While this is generally acceptable, if you experience slowdowns or hanging when DeepSpeed exits, it is recommended to set the TRITON_CACHE_DIR environment variable to a non-NFS path.
 [WARNING]  async_io requires the dev libaio .so object and headers but these were not found.
 [WARNING]  async_io: please install the libaio-dev package with apt
 [WARNING]  If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
INFO 06-25 10:57:10 [importing.py:53] Triton module has been replaced with a placeholder.
INFO 06-25 10:57:10 [__init__.py:239] Automatically detected platform cuda.
[INFO:swift] Currently, rollout only supports the vLLM backend. Set vLLM backend
[INFO:swift] currently, rollout don't support async engine, set use_async_engine False
[INFO:swift] Downloading the model from ModelScope Hub, model_id: Qwen/Qwen2.5-VL-3B-Instruct
[WARNING:modelscope] Using branch: master as version is unstable, use with caution
[INFO:modelscope] Target directory already exists, skipping creation.
[INFO:swift] Loading the model using model_dir: /bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2___5-VL-3B-Instruct
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
[INFO:swift] Setting image_factor: 28. You can adjust this hyperparameter through the environment variable: `IMAGE_FACTOR`.
[INFO:swift] Setting min_pixels: 3136. You can adjust this hyperparameter through the environment variable: `MIN_PIXELS`.
[INFO:swift] Setting max_pixels: 12845056. You can adjust this hyperparameter through the environment variable: `MAX_PIXELS`.
[INFO:swift] Setting max_ratio: 200. You can adjust this hyperparameter through the environment variable: `MAX_RATIO`.
[INFO:swift] Setting video_min_pixels: 100352. You can adjust this hyperparameter through the environment variable: `VIDEO_MIN_PIXELS`.
[INFO:swift] Setting video_max_pixels: 602112. You can adjust this hyperparameter through the environment variable: `VIDEO_MAX_PIXELS`.
[INFO:swift] Setting video_total_pixels: 90316800. You can adjust this hyperparameter through the environment variable: `VIDEO_TOTAL_PIXELS`.
[INFO:swift] Setting frame_factor: 2. You can adjust this hyperparameter through the environment variable: `FRAME_FACTOR`.
[INFO:swift] Setting fps: 2.0. You can adjust this hyperparameter through the environment variable: `FPS`.
[INFO:swift] Setting fps_min_frames: 4. You can adjust this hyperparameter through the environment variable: `FPS_MIN_FRAMES`.
[INFO:swift] Setting fps_max_frames: 768. You can adjust this hyperparameter through the environment variable: `FPS_MAX_FRAMES`.
[INFO:swift] default_system: 'You are a helpful assistant.'
[INFO:swift] max_length: 128000
[INFO:swift] response_prefix: ''
[INFO:swift] agent_template: hermes
[INFO:swift] norm_bbox: none
Downloading Model from https://www.modelscope.cn to directory: /bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2.5-VL-3B-Instruct
INFO 06-25 10:57:22 [config.py:717] This model supports multiple tasks: {'embed', 'generate', 'reward', 'classify', 'score'}. Defaulting to 'generate'.
INFO 06-25 10:57:22 [config.py:2003] Chunked prefill is enabled with max_num_batched_tokens=5120.
[2025-06-25 10:57:27,693] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:49: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(ctx, input, weight, bias=None):
/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/deepspeed/runtime/zero/linear.py:67: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
[INFO:swift] Successfully registered `/bd_byt4090i1/users/jhu/miniconda3/envs/robot-vlm/lib/python3.10/site-packages/swift/llm/dataset/data/dataset_info.json`.
Warning: The default cache directory for DeepSpeed Triton autotune, /home/nfs/jhu/.triton/autotune, appears to be on an NFS system. While this is generally acceptable, if you experience slowdowns or hanging when DeepSpeed exits, it is recommended to set the TRITON_CACHE_DIR environment variable to a non-NFS path.
 [WARNING]  async_io requires the dev libaio .so object and headers but these were not found.
 [WARNING]  async_io: please install the libaio-dev package with apt
 [WARNING]  If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
INFO 06-25 10:57:30 [importing.py:53] Triton module has been replaced with a placeholder.
INFO 06-25 10:57:30 [__init__.py:239] Automatically detected platform cuda.
INFO 06-25 10:57:32 [core.py:58] Initializing a V1 LLM engine (v0.8.5.post1) with config: model='/bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2___5-VL-3B-Instruct', speculative_config=None, tokenizer='/bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2___5-VL-3B-Instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.bfloat16, max_seq_len=128000, download_dir=None, load_format=auto, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=True, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='auto', reasoning_backend=None), observability_config=ObservabilityConfig(show_hidden_metrics=False, otlp_traces_endpoint=None, collect_model_forward_time=False, collect_model_execute_time=False), seed=None, served_model_name=/bd_byt4090i1/users/jhu/modelscope/models/Qwen/Qwen2___5-VL-3B-Instruct, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=False, chunked_prefill_enabled=True, use_async_output_proc=True, disable_mm_preprocessor_cache=False, mm_processor_kwargs=None, pooler_config=None, compilation_config={"level":3,"custom_ops":["none"],"splitting_ops":["vllm.unified_attention","vllm.unified_attention_with_output"],"use_inductor":true,"compile_sizes":[],"use_cudagraph":true,"cudagraph_num_of_warmups":1,"cudagraph_capture_sizes":[512,504,496,488,480,472,464,456,448,440,432,424,416,408,400,392,384,376,368,360,352,344,336,328,320,312,304,296,288,280,272,264,256,248,240,232,224,216,208,200,192,184,176,168,160,152,144,136,128,120,112,104,96,88,80,72,64,56,48,40,32,24,16,8,4,2,1],"max_capture_size":512}
INFO 06-25 10:57:32 [worker_base.py:589] Injected <class 'trl.scripts.vllm_serve.WeightSyncWorkerExtension'> into <class 'vllm.v1.worker.gpu_worker.Worker'> for extended collective_rpc calls ['close_communicator', 'init_communicator', 'update_named_param']
WARNING 06-25 10:57:32 [utils.py:2522] Methods determine_num_available_blocks,device_config,get_cache_block_size_bytes,initialize_cache not implemented in <vllm.v1.worker.gpu_worker.Worker object at 0x7f18985d35e0>
INFO 06-25 10:57:32 [parallel_state.py:1004] rank 0 in world size 1 is assigned as DP rank 0, PP rank 0, TP rank 0
INFO 06-25 10:57:32 [cuda.py:221] Using Flash Attention backend on V1 engine.
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.

**Comment by hjh0119** [comment]
[WARNING] async_io requires the dev libaio .so object and headers but these were not found.

maybe install libaio-dev
